cou/utils/openstack.py

```diff
@@ -464,3 +409,48 @@ def is_charm_supported(charm: str) -> bool:
     :rtype: bool
     """
     return bool(OpenStackCodenameLookup.lookup(charm)) or charm in SUBORDINATES
+
+
+def _generate_track_mapping() -> dict[TrackKeys, str]:
+    """Generate the track mapping for the auxiliary charms.
+
+    This mapping should be updated periodically by adding new lines in the file
+    openstack_to_track_mapping.csv
+
+    See the following url for more details:
+    https://docs.openstack.org/charm-guide/latest/project/charm-delivery.html
+
+    :return: Dictionary containing the tracks by charm name, series and OpenStack release.
+    :rtype: dict[namedtuple, str]
+    """
+    track_mapping = {}
+    with open(
+        Path(__file__).parent / "openstack_to_track_mapping.csv",
+        encoding=encodings.utf_8.getregentry().name,
+    ) as csv_file:
+        csv_reader = csv.DictReader(csv_file, delimiter=",")
+        for row in csv_reader:
+            track_key = TrackKeys(
+                charm=row["charm"], series=row["series"], os_release=row["os_release"]
+            )
+            track_mapping[track_key] = row["track"]
```

Comment:
Not suggesting to any change. My original idea was to define this part directly in python e.g.
```python
OPENSTACK_TO_TRACK_MAPPING = {
    TrackKeys("ceph-mon", "focal", "ussuri"): "octopus",
    ...
}
```